### ISSUE TITLE
Feature request

### ISSUE THREAD

**Issue by ArtemMedvedev** [BoostIO/Boostnote#125]
Thanks for awesome app, exactly what I was looking for.

I might have miss, but for the first look haven't found this in issues. 
It would be cool have possibility to make nested folder structure, adding subfolders.
Also, default syntax for folder and opportunity to edit syntaxes list (hide unused), would accelerate and simplify adding new notes.
And lastly, would be very useful have more options for customizing Finder. Search by tags, change font, theme, sizes, hide sidebar, etc.

Everything else has already been said in the other threads.
I am very happy that found Boostnote, it will be very actively used.

**Comment by Rokt33r** [comment]
Thank you for your feedback.

> It would be cool have possibility to make nested folder structure, adding subfolders.

We need to discuss this issue more. How many notes you have? And, I'll make a new issue about it later.

> default syntax for folder and opportunity to edit syntaxes list (hide unused)

This acutally is actually planned. I think It will be shipped within 2 weeks

> lastly, would be very useful have more options for customizing Finder. Search by tags, change font, theme, sizes, hide sidebar, etc.

It would be great. But, all of problems is how it should be.(For me, designing app is MUCH HARDER than writing code) Could you give me more information here?  #85

**Comment by ArtemMedvedev** [comment]
@Rokt33r Hey! Sorry for delay, a little bit busy this weekend.

> We need to discuss this issue more. How many notes you have? And, I'll make a new issue about it later.

Currently I have approximately 170+ code snippets which I want to keep structured by categories, for a quick example:

JavaScript:
- NodeJS
- ES6
- React
- Electron
- Gulp
- ...

DevOps:
- Nginx
- Docker
- Varnish, Memcached
- ...

AppleScript:
- General
- Apps
- Finder
- Script Debugger
- ...

Shell scripts:
- Bash
- Logs
- Vim
- ZSH, Prompts
- ...

Automation:
- Alfred
- Hazel
- Keyboard Maestro
- ...

and so on.

Also I have about 400+ bookmarks in the [raindrop](https://raindrop.io). I really like it. but it's absolutely not suitable for storage code snippets. I tried to use SnippetsLab, Dash, GistBox, CodeBox, and others. In most cases opportunities are very limited, I can't feel like in the editor/IDE. Even such simple things as make a comment on the usual hotkey, unfortunately almost impossible anywhere, about a deeper customization can only dream of, and usually support with updates always leaves much to be desired.
In addition to all I want to store personal library of markdown docs, different tech manuals and hotkeys cheatsheets, for example be able make a quick edits in them to use it with [cheaters](http://ttscoff.github.io/cheaters/). Now for this I use [MWeb](http://www.mweb.im/), but I feel not a very convenient to constantly switch between different apps, keep them always running, remember different hotkeys for the same things, I think it isn't right when this kind of apps, doesn't have ability to change hotkeys, or at least doesn't follow the usual macOS hotkeys (I'm sure there are good reasons).
From time to time I'm doing the cleaning in the snippets, bookmarks and other relevant stuff, but over time, the amount of material which I would like to keep only growing. And finally, I think I found what will eventually become the **instrument of dreams** for this. I have a lot of material but it's scattered absolutely everywhere where only it's possible, want to put everything in order.

> default syntax for folder and opportunity to edit syntaxes list (hide unused)
Cool, this is a great news!

> It would be great. But, all of problems is how it should be.(For me, designing app is MUCH HARDER than writing code) Could you give me more information here? #85 

I need to think how to do it better, make a small analysis of personal preferences and observations, highlight the best things from all that I already tried, and of course how this may be the most useful and convenient for all. I'll write in #85 in a few days.

**Comment by Rokt33r** [comment]
Great! Thanks you so much! Your feedback is so helpful.

I am acutally going to several ways to keep and indexing notes by the purpose like #123 .

> Even such simple things as make a comment on the usual hotkey, unfortunately almost impossible anywhere, about a deeper customization can only dream of, and usually support with updates always leaves much to be desired.

Let's do it.

I think these api should be provided customizable hotkey. And also I think i should provide some scope of hotkey.(It might be optional)

|Api|key|scope|
|---|---|---|
|New note(default behaviour, same as clicking the plus button)|Cmd + N(OS X) / Ctrl + N(Others), `a`(when focusing SideNav)|global|
|New Snippet note|Cmd + N S / Ctrl + N S|global|
|New Markdown note|Cmd + N M / Ctrl + N M|global|
|New note dialog|Cmd + N N / Ctrl + N N|global|
|Delete note|Ctrl + Backspace(OS X) / Ctrl + Delete(Others)|global(except focusing editor)|
|Focus detail|`e`|side nav and list|
|Focus editor|?|detail and list|
|Focus tag input|?|detail|
|Focus filename input|?|detail|
|Focus syntax mode input|?|detail|
|Add Favorite|Cmd + D / ?|global|
|Focus search input|Cmd + P / ?|global|
|Change font size of renderer|Cmd + `+`(OS X)|detail|
|Focus sidebar(to navigate folders and storages by arrow keys, `↑` and `↓`)|list and detail|
|Create folder|Shift + `A`|side nav|
|Delete folder|`d`|side nav|
|Add Storage|?|side nav|
|Unlink Storage|?|side nav|
|Rename Storage|?|side nav|
|Focus Note List(to navigate notes by arrow keys, `↑` and `↓`)|Esc|sidenav and detail|

`?` is I cannot sure which key would be proper. I think some of them doesn't need to be provided by default.
If I missed something or did wrong, please tell me.

> I want to store personal library

It is so interesting. I think if we provide a storage like Wiki, it would be helpful for building it.

By the way, I want you to join our slack. If you don't mind, send me an e-mail and I'll invite you. fluke8259@gmail.com
It is integrated with this storage. So, you could check our progress easily.

I'm look forward to your next feedback. I'm glad to help you realize your ideal. 👍

**Comment by UltraG33k** [comment]
Hi! Lovely app.
I also have feature requests:
An option for the following
    - Start minimized when system starts
    - Start normal when system starts
    - And of cause not starting up at system start

It would be nice to not have to make fixes to do so.
Thanks in regards :)

**Comment by Rokt33r** [comment]
@UltraG33k 
We might be able to implement it. Thanks for your reporting.
https://www.npmjs.com/package/auto-launch

**Comment by Rokt33r** [comment]
@ArtemMedvedev 
We decided to implement the nested structure of folders. 👍  I'll inform you the detail later.

**Comment by frtru** [comment]
Any news on the nested struct of folders feature? Looked at the changelog and PRs, but I can't seem to find anything..

**Comment by pat-s** [comment]
Push the "nested folder" idea!

**Comment by oicitrapdraz** [comment]
No news yet? It would be great to be able to create subfolders, otherwise it's difficult to organize so many notes.

**Comment by ctoscano** [comment]
@Rokt33r Hi. Thanks for your contributions. Any news on supporting subfolders?

**Comment by Rokt33r** [comment]
We're going to renew Boostnote soon. And, definitely, the renewed app should have sub folders.

**Comment by ctoscano** [comment]
Wohoo. Thanks for the update @Rokt33r. Is there a branch I could follow? I'd be curious to learn how a feature like that would come to exist.

**Comment by Rokt33r** [comment]
@ctoscano 
Not available yet. I'm going to quit my job at the end of this year to focus on this project. I'll address it when I'm ready to go!

**Comment by mehrdaad** [comment]
Hey @Rokt33r sorry for pinging again, when do you think this feature will be available?

**Comment by danvayn** [comment]
Excellent Notetaking app for developers but I'm still waiting on subfolders for note organization

**Comment by corept** [comment]
I'm also very interested in a sub-folder feature. Hopefully coming soon.

**Comment by invacuo** [comment]
Agree with @orlando. Bear notes supports nested tags and is super easy and convenient to organize my notes.

**Comment by thecountoftuscany** [comment]
Thanks for the great app! I am also looking forward to seeing the nested folders feature implemented.

**Comment by zmrhaljiri** [comment]
The best MD notes for Windows :) Please, nested folders are just one thing that's missing to the 100 %.

**Comment by Leiknir** [comment]
Any news about this? For me, this is the only feature missing that would make boost note as the primary app for managing my notes. Thank you.

**Comment by P0oOOOo0YA** [comment]
Can't believe such a **fundamental** feature has not been implemented yet.

**Comment by orlando** [comment]
You are free to implement it @P0oOOOo0YA

**Comment by liam-b** [comment]
Seriously considering switching from Evernote to this, but subfolders are a vital part of my workflow!

**Comment by FreddyCrugger** [comment]
No subfolders yet? Wow this is a very basic feature that kills the whole snippet idea, sad.

**Comment by Dreamer3783** [comment]
Subfolders — forever! I fully pass on to you from Nimbus Note, when you organize subfolders. Thank you for your app. You — better and the best!

**Comment by aakbar5** [comment]
any update on this feature request?

**Comment by lanox** [comment]
Hi any update on subfolders please.

**Comment by Flexo013** [comment]
Currently it does not look like anybody is working on implementing subfolders.

If you would really like this feature, then consider funding it on [IssueHunt](https://issuehunt.io/r/BoostIo/Boostnote). That way developers are incentivized to pick up this feature request.

**Comment by FreddyCrugger** [comment]
It is ok thanks, I am thinking about building something similar very soon with Electron for people to download for free. I think it is needed. Should not be too difficult.

**Comment by Flexo013** [comment]
There were many ideas thrown around in this thread/issue, but it's hard for these to be picked up by devs when they are scattered through a long thread.

Let's make this issue focused on "Subfolders" as they seem the most popular of the features.

@artdevjs  @UltraG33k @orlando 
Feel free to open separate suggestions/issues for the other features that you suggested!

**Comment by orlando** [comment]
Sorry this took so long that I switched to Notion. I won’t be able to work on this.

**Comment by Flexo013** [comment]
@orlando Understandable.

**Comment by WeslleyNasRocha** [comment]
I will try to tackle on the subfolder issue

**Comment by WeslleyNasRocha** [comment]
I will try to look on it

**Comment by AntowaKartowa** [comment]
As a workaround it is possible to make folders in your Dropbox and add them as additional storage.

**Comment by Ghost-Writter** [comment]
yeah, i want subfolders for using that app. it's should be vital.

**Comment by dirathea** [comment]
Hi guys, Currently I tried to start working on this features, here's my work in progress https://github.com/BoostIO/Boostnote/pull/3288. This is my first PR on this repo and I hope I could give a little contribution 🙇 

Feel free to give comments and suggestions for my work.

Thanks

**Comment by Purukitto** [comment]
Can I work on this?

**Comment by Flexo013** [comment]
@Purukitto Yes you can, it's also worth checking out the current PR #3288 and see if there's anything that can be improved for that implementation.

**Comment by pabrams** [comment]
Hello, I'm confused... it looks like Boost Note already has this feature: 
![image](https://user-images.githubusercontent.com/1376688/145519438-f94c332e-1743-4cb1-84f1-1a8a5d319b30.png)
Also, all these issues and bounties are still open, but it looks like there hasn't been any activity in a while. Is that why the name of the repo has "legacy" in it? What does that mean, if the bounties are still open? @Flexo013 @Rokt33r
